hack/jenkins/e2e.sh

```diff
@@ -111,6 +111,7 @@ GKE_REQUIRED_SKIP_TESTS=(
     "MasterCerts"
     "Shell"
     "Daemon\sset"
+    "Deployment"
```

Comment:
The comment is technically still accurate, as at least some of the other tests are skipped because they require ssh access to the master. Updated anyway, filed #14751.